### ISSUE TITLE
Multiple bugs

### ISSUE THREAD

**Issue by SIREN587** [OpenApoc/OpenApoc#822]
1. Sometimes craft that goes to mission,after mission ends the craft stays in that building and does not return to base

2. Raiding buildings(as in cult of sirius) dosent give you the weapons and items looted.

3. Some organizations may have aliens in their buildings despite ufopedia and ufograph saying 0%

4.When killing popper with grenade game crashes. Especially if the popper is on top of desks-book cases etc.

5. Some aliens for whatever reason stay inactive,they dont move. Enable bughunt after round10 eg the aliens agressively search you or they leave fight and you win. Nobody likes to play hide and seek. Openxcom has this feature(bughunt)

6. Some weapons say 60% accuracy. This SHOULD MEAN that 6 out of 10 times a hit should connect. Otherwise change it to 40% to accurately display whats going on.

7. When the game runs at maximu speed, when ufos appear game doesnt pause 

8, When ufos do attack missions(eg they attack various buildings) the relationship of that organization doesnt go hostile or even reduce against aliens,if xcom attacks building relations go hostile right away

9. when ufos attack a building that buildings forces rarely if ever retaliate or send aircraft.Even megapol doesnt do anything :D

10. This is the best bug ever, If alien ship goes to alien dimension before your fighters shoot at it,your fighters start attacking the city oO

11. aliens not visible well if they are on top of items. maybe add permanent marker to know that you saw an alien somewhere

12. Vehicle weapons damage is too low,you cannot deal with 3 fast attack ships and one bomber and one destroyer on superhuman.

13. absolutely zero raids between organizations.ZERO RAIDS whatsoever

14. actually the above bug with your fighters attacking the city no matter what persists.Quite funny actually 

15. Aircraft weapons say 60-65% accuracy. Its more like 40-45%. Either change ufopaedia or buff accuracy. 65% means 65%

16. Anthropods dont use grenades for whatever reason

17. Skeletoids stay inactive or hide instaed of attacking.Especially when one is left

18. The big worm whhen killed drops 4 mini worms. The idea behind this unit was that when you managed to kill this unit to be faced with 4 more,so you had to neutralize it via gas. Here when you kill it the 4 new worms insta kill you :DDDDDDDDDDDDDDDDDDDDDDDDDDDDD 

19. visibility is bad,you may pass near an alien 50 times but unless you go straight to him you will not see him

20. Pathfinding is really bad.Way worse than the original game

21. zero auto equip after each battle,maybe make it so atleast equiping is easier just like in openxcom 1 click and grenade goes to belt eg

22. lel power lel sword lel not lel working lel out lel of lel range lelelelelelelelelelel top kek. 

23. if you build mega workshop old engineers cannot be used, you need to buy new ones :D

24. research tree is broken,some items dont appear at all eg devastator cannon

25. if you dont down ufo number 3 when its appearing you may lose it forever :S 

26.  after a while the aircaft you use is not available to change modules

I could write three times more the bugs encountered by i ll stop here just in case this project has stopped. 

latest openapoc build used with gog game latest

[save_SAVE3.zip](https://github.com/OpenApoc/OpenApoc/files/4439010/save_SAVE3.zip)

**Comment by SIREN587** [comment]
![WTF32](https://user-images.githubusercontent.com/63198187/78573928-f8231200-7831-11ea-87b2-e108da9cefc6.jpg)


different bug again with popper and grenade

**Comment by FilmBoy84** [comment]
Thanks for bug-list @SIREN587 
This is very useful

A few are known issues, but more details are always welcome

1) Are you playing with "Raiding destroys building" on? If so, the craft will wait until leaving the garage in the building so as not to have tiles come crashing down on it

2) With regards to 2. I cannot replicate this, do the vehicles you send on a raid have a cargo module? (they need to to collect items)

3) This is normal. and in the original game it was the same, a building can house aliense without the parent organisation being infiltrated. Infiltration is a level of alien control - not just where they are sitting around (though that does influence it)

5) The unitAI is not yet implemented. There is a LOT of behaviour to add such as searching, false flags, retreating, call for reinforcement, grouping, leaving the combat area, strafing, crawling, seeking cover, etc.

6) There are some accuracy bugs that need resolving now we have tied things to 60fps/tps as the accuracy algorithm is tied to both ticks and frames. However, your assessment that 60% accuracy means 6/10 shots hit is incorrect, 60% accuracy means that if your agent has a 100 accuracy stat then EACH shot will have a 6/10 chance of hitting. Agents with worse accuracy skill will have worse than 60% accuracy with that weapon. This is intentional and the same as OG.

7) Interesting, sometimes it does, sometimes it doesn't, thanks for report. We need to make a separate issue for this (and most of these others if SEARCH reveals nothing)

8) Relations are not fully implemented yet, currently OpenApoc only uses one value to measure relations between each party (two values per relationship one to the other and vice versa) the original game uses two for each side in a relationship (short term changes, long term bias) or a total of four values in each two way relationship

9) Because relations are not correctly implemented yet, the effects of attacking a building are "irregular" in terms of the parent organisations response.

10) Can you please replicate this in a video and make a new issue solely for that bug if a search does not reveal an existing report?

11) This is part of several known LoS bugs #356 #360 

12) Damages are taken from the original game, there are some bugs relating to TPS/FPS linked calcs we are resolving (as per above) but not being able to down UFOs alone is intentional

13) Raids do not function as moving organisation/alien units between buildings is not implemented yet

15) Again, accuracy is not a clean "65% means 65%". Lots of variables affect accuracy in X-COM this is the case in UFO/TFTD/Apoc and is why "X-COM Math" is a thing. To have full 65% accuracy you have to have agent/pilot/driver/craft stats of 100 and be taking a shot against a stationary target within the optimal range of the weapon. Skill, Movement, Distance, Speed, all of these play a part in deviating the accuracy from that quoted (much as in real life)

16 & 17) This again is down to UnitAI not being implemented

18) Hyperworms are, and have always been, extremely deadly, try and kill them with HE or Plasma/Disrupter Spam before they get in C/C range

19) Line of Sight is a known issue #356 #360

20) Pathfinding is not fully implemented - Agents don't move around other agents or wait for agents to move like OG - they just give up, this is a known issue

21) Auto-refill ammo and replace used equipment not occuring is a known issue. HOWEVER, for now you can set equipment templates with Ctrl-#NUMBER# when you press that number again on anther agent, it will equip to the spec set (a feature not in OG)

22) Powersword has a range of 1 tile, can you please make a seperate issue and a video of this not working so we can replicate please?

23) Engineers and Scientists causing all manner of assignment and economy bugs is a known issue, thanks for further info

24) I cannot replicate this... Research Tree works for me, can you please post a list of all you have researched so we can find any holes?

25) UFO3 not reappearing, or players downing it in the alien dimension and never being able to recover is an issue with the OG... And it's intentional. Apoc was, at one time, full of intentional ways to mess up player progression (a legacy of the one way to win concept) many of these were patched over before release, but UFO growth was never fully fixed. For now, OpenApoc copies the original game. We'll add an option or mod to change things eventually

26) I cannot replicate this, can you please create a seperate issue and add a video and savegame?

Many thanks for all of these, it must have taken some time to put together, so i thank you

One request for the future though; can you please put each issue as a seperate report - the problem with these multiple issue reports is they get buried and are hard for our voluntary devs to track properly

Kind Regards, and don't worry, development is still on-going

FB84

**Comment by SIREN587** [comment]
2. Yes i have module, i check my gear in base,go raid,check base gear again its the same numbers.Storage is emtpy. I went to raid cult of siriius and this time i took some items in my inventory.When i returned the items where in the backpack and i could put them in storage base

3. are you sure? Because after a while i get the screen x is now with aliens but this is not the end etc etc. 

9. i dont know if i can video this but i usually use lot of saves,next time the aliens attack i will monitor. and copy paste the save here again. 

22 powersword in the original game had a range of 2. In this games the range is 1. But when you attack an enemy it says enemy is out of range oO only once i managed to hit an enemy diagonally and afterwards its always out of range. I will try to use it again


Btw save in first post has all these issues. It also has the issue of broken reasearch tree and what not. No idea where the game hiccuped.

**Comment by FilmBoy84** [comment]
I noticed you are using the GOG version?

Can you do me a favour and check if your GOG install has, under the UFOEXE directory and within the CUE/BIN CD-ROM image as well

UFO2P.EXE (1,702,206 bytes)
UFO2P4.EXE (1,705,790 bytes) 

Please make sure that UFO2P.EXE is also the correct size and not just a renamed UFO2P4.EXE

There was a problem identified last week with the gog release i put a ticket and forum post in about here

https://www.gog.com/forum/xcom_series/xcom_apocalypse_serious_distribution_bug_please_fix_gog

With regards to 3) as i also said above, relations are a mess and not fully implemented
So you have a double whammy of Infiltration mechanics not being properly finished yet and relations not being fully finished.... This leads to some strange behaviours...

**Comment by SIREN587** [comment]
Actually i have only ufo2p.exe at 1,705,790 bytes and no ufo2p4.exe. 

As for the relations mechanics,the game back then,according to gollop at least, tried to randomize various factions, so no two games would be alike, eg in one game you would be allies with psyke in anotther with osiron etc etc. Each would raid you. Instead nothing of this was implemented.Even cult of sirius rarely raids you.

He also said something about missions in the alien dimension which never came to be. From 1994 onwards came the slow death of pc gaming.But anyway thanks for doing this, xcom was one of the few great games. I will try to replicate some bugs later on if possible

**Comment by Rezistor87** [comment]
Yes, i have GOG version, same bugs.

![image](https://user-images.githubusercontent.com/63288674/78652887-9e1e5d00-78c2-11ea-90ef-c56a0d07bb49.png)

After this one, time boost on the map is totaly frozen.

**Comment by SIREN587** [comment]
I played the gog version alone,without openapoc.

The game has the following bugs, 

1.during real time some times the missions doesn't end even though all enemies are dead

2, enemies that might be near your to the left or right might not be visible unless you go right at them which is totally fantastic, humans have peripheral vision of 180 degrees etc

3. game might crash for no reason at all filling the whole screen with red letters-code(although this part might be due to dosbox)

4. Game after week 15? doesn't send ufos at all even though no building is broken in alien dimension

5. rarely ai uses overspawn and the one time it used it it barely broke 2 buildings before being killed.

6. apocalypse mission does not proc( the one that 100s of ufos come to attack the city after you *spoiler* "

7. zero raids so far from opponents(cult of sirius) even though i raided them over 10 times in a row.Also destroying all their buildings havent proced a raid.

All the above on superhuman difficulty if it matters.

For a gog game i thought that it was a repack with latest patches yada yada.Instead the game has more bugs it seems. If by any chance i get my hands on a steam version i will update here so as to not clatter this board with more threads. If anyone is reading this avoid using the gog version due to bugs.Have a nice day all

**Comment by FilmBoy84** [comment]
With regards to the GOG version and OpenApoc the EXE issue shouldn't matter (all the code is our own, we don't use the EXEs)

But if one thing on the GOG release is a mess, i suspect there is more wrong with it.

The problems have only arisen since Gog Galaxy 2 hit and i have an unnerving feeling the GOG distro was corrupted and someone just renamed UFO2P4.exe to UFO2P.exe and went "it works"

I never had problems with the GOG version of Apoc until quite recently now every game seems to break eventually unless i put the steam EXEs into the GOG distro and start a new game

Gah, i'm not sure how we resolve these errors with OpenApoc when we don't know if it's the GOG distro behind them

@JonnyH are you able to advise please?

Common sense says, can we replicate all of this with the Steam version ISO? But im not sure if SIREN587 has a copy of that for testing and it would also require starting a new game in OpenApoc
